### ISSUE TITLE
warning: 'AuthorizationExecuteWithPrivileges' is deprecated: first deprecated in OS X 10.7

### ISSUE THREAD

**Issue by 100ideas** [codebutler/firesheep#209]
Apparently AuthorizationExecuteWithPrivileges() is deprecated in OS X 10.7. Everything compiles (with warnings about this deprecation) but when running the extension in Firefox 25, after clicking "Start Capturing" and authenticating, an error is thrown 

`osx_run_privileged: AuthorizationExecuteWithPrivileges() failed: -60031`

The deprecated code is in [osx_platform.cpp](https://github.com/codebutler/firesheep/blob/d2645fee156b267ca24b31e452feaff5c8c5e957/backend/src/osx_platform.cpp#L48).

More info about this on [stack overflow](http://stackoverflow.com/questions/6841937/authorizationexecutewithprivileges-is-deprecated).

**Comment by tsujamin** [comment]
yeah I encountered this last night as well on OSX10.9/FF25. Going to have a play with it today to see what I can do..

EDIT: yeah seems to occur regardless of where the backend binary is placed/executed. 
If sudo firesheep-backend --fix-permisions is called from a prompt then the error is avoided. Is this experienced on all versions of OSX currenty?

**Comment by tsujamin** [comment]
well thats strange, just got a friend to build the latest revision on his 10.8 machine and his version doesnt suffer the -60031..

**Comment by 100ideas** [comment]
I have FileVault turned on, could that be it?

—

**Comment by tsujamin** [comment]
I run FileVault also... The computer my friend used to build the working version (OSX 10.8, which happened to work on my machine) didnt have it enabled...

**Comment by furkanayhan** [comment]
I use Mavericks, and I get these errors:

![ekran resmi 2013-12-10 14 21 41](https://f.cloud.github.com/assets/1630249/1713891/bf780a3c-6195-11e3-990b-4d50a4f06688.png)
![ekran resmi 2013-12-10 14 21 20](https://f.cloud.github.com/assets/1630249/1713892/bf931f48-6195-11e3-93d5-e2b9e41f705f.png)

**Comment by codebutler** [comment]
You can work around this issue by running:

```
$ sudo /tmp/firesheep/firesheep-backend --fix-permissions
```

I'll try to figure out a better solution, but it seems the new official API requires codesigning.

https://developer.apple.com/library/mac/samplecode/SMJobBless/Introduction/Intro.html#//apple_ref/doc/uid/DTS40010071-Intro-DontLinkElementID_2

This thread has some truly awful hacks involving applescript. I'd be interested to know what that actually does under the hood.
http://stackoverflow.com/questions/6841937/authorizationexecutewithprivileges-is-deprecated

**Comment by tsujamin** [comment]
Just a curious sidenote: I built a copy on 10.8, installed it onto my 10.9 machine and it worked. I did read that section about helper tools and codesigning, seems like alot of hoops to jump through

**Comment by furkanayhan** [comment]
@codebutler So, what should I do to compile firesheep successfully?
